x/finality/keeper/gov.go

```diff
@@ -117,5 +126,29 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 		k.SetVotingPowerDistCache(ctx, h, distCache)
 	}
 
+	// it is possible that some inactive fps become active after the proposal
+	// therefore, we need to ensure every active finality provider has signing info
+	for pk, dc := range distCache.GetActiveFinalityProviderSet() {
```

Comment:
doesn't this fix work only for last `distCache` we iterated in the above loop i.e after above loop `distCache` pointer will point to last modified `VotingPowerDistCache`

This means that if some finality providers were activated in the intermediate distcaches, problem will persist for them.

Reply:
Ah good point. We probably need to accumulate all the active fps from halting height to current height

Reply:
hmm I do not think so  🤔 we evaluate liveness after  `FinalitySigTimeout` i.e
```go
		heightToExamine := sdk.UnwrapSDKContext(ctx).HeaderInfo().Height - k.GetParams(ctx).FinalitySigTimeout

		if heightToExamine >= 1 {
			k.HandleLiveness(ctx, heightToExamine)

```

Reply:
You are right. In a case where an fp becomes active at `current_height -1` and then inactive at `current_height`, we will miss it. Maybe we do a conservative fix by accumulating all the active fps and ensure they have signing info

Reply:
Done in [853cd05](https://github.com/babylonlabs-io/babylon/pull/860/commits/853cd05346869cae1cb038c2a163349f7f0673bf)

Reply:
ahh okay, here you are going to use `allActiveFps`, I think it is missing to replace 

```go
	for pk, dc := range distCache.GetActiveFinalityProviderSet() {
```

by

```go
	for pk, dc := range allActiveFps {
```

Reply:
Sorry, just updated the commit hash. The local change was not pushed correctly